Add initial logic (#3)
canonical/kratos-external-idp-integrator: README.md

## Description

This charm is used to configure an Ory Kratos charm to use an external provider.

## Usage

### Client registration

> Before deploying this charm you should go and register an oidc client with the provider you wish to use. Instructions for
> registering a client for some well known provider can be found below. Instructions for a larger list of providers can be found at the
> Ory Kratos [docs](https://www.ory.sh/docs/kratos/social-signin/overview)

Reviewer: ```suggestion
Before deploying this charm you should register an oidc client with the provider you wish to use. Instructions for
registering a client for some well known providers can be found below. Instructions for a larger list of providers can be found at the
Ory Kratos [docs](https://www.ory.sh/docs/kratos/social-signin/overview).
```

> Note that after registering a client you will need to provide a redirect_uri to the provider, you will get the redirect_uri
> after deploying the integrator.

Reviewer: ```suggestion
Note that after registering a client you will need to provide a redirect_uri to the provider. It can be fetched
once the integrator charm is deployed.
```

#### Azure AD

Instructions for registering a client on Azure AD can be found [here](https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app).

#### Okta

> Instructions for registering a client on Azure AD can be found [here](https://developer.okta.com/docs/guides/find-your-app-credentials/main/).

Reviewer: ```suggestion
Instructions for registering a client on Okta can be found [here](https://developer.okta.com/docs/guides/find-your-app-credentials/main/).
```

### Deployment

> For the kratos-external-idp-integrator charm to operate you need to deloy it, configure it and relate to a kratos charm.:

Reviewer: ```suggestion
For the `kratos-external-idp-integrator` charm to be operative you need to deploy it, configure and relate to the kratos charm:
```

```commandline
juju deploy kratos-external-provider-integrator

> juju config kratos-external-provider-integrator ...

Author: I wasn't sure if documenting the configurations here was necessary, because charmhub will autogenerate docs about the config using `config.yaml`. I looked into other charms and it seems like most of them don't do this.

However if you feel like adding a couple of words about the configurations will make things clearer let me know.

> juju config kratos-external-provider-integrator ...

Reviewer: I wasn't aware of the autogenerated docs. It would be nice to have info about required configs or link to the charmhub documentation
